### PR TITLE
Transform: Add a limit transform

### PR DESCRIPTION
**Pull request by josiahg**
From a user:

_We have a Table panel where we want to limit the number of entries displayed to the top ten items based on aggregation transforms that happen on the Grafana side (therefore we cannot use the "Limit" for the data source; that skews the source data prior to aggregation). This seems like it should be a very common option for the Table visualization, but there doesn't appear to be an obvious option or transform to accomplish this._

_What was expected to happen?
The table should be truncated after the set number of items is displayed._

_Because the counts will vary by the time period selected, there’s no good cutoff that would allow use of the Filter Data by Value transform to limit the list to a certain number of entries._

This PR adds just such a transform, allowing you to limit the number of query results displayed to a given number.

## Limit

Use this transformation to limit the number of rows displayed.

In the example below, we have the following response from the data source:

| Time                | Metric      | Value |
| ------------------- | ----------- | ----- |
| 2020-07-07 11:34:20 | Temperature | 25    |
| 2020-07-07 11:34:20 | Humidity    | 22    |
| 2020-07-07 10:32:20 | Humidity    | 29    |
| 2020-07-07 10:31:22 | Temperature | 22    |
| 2020-07-07 09:30:57 | Humidity    | 33    |
| 2020-07-07 09:30:05 | Temperature | 19    |

Here is the result after adding a Limit transformation with a value of '3':

| Time                | Metric      | Value |
| ------------------- | ----------- | ----- |
| 2020-07-07 11:34:20 | Temperature | 25    |
| 2020-07-07 11:34:20 | Humidity    | 22    |
| 2020-07-07 10:32:20 | Humidity    | 29    |

### CODE REVIEW

**Comment by josiahg** [comment]
@ryantxu I'd love your feedback/review on this one!  Current state: it works, but code may be iffy?

File: packages/grafana-data/src/transformations/transformers/limit.ts [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+  description: 'Limit the number of items to the top N',
+  defaultOptions: {
+    limitField: DEFAULT_LIMIT_FIELD,
+  },
+
+  operator: (options) => (source) =>
+    source.pipe(
+      map((data) => {
+        const limitFieldMatch = options.limitField || DEFAULT_LIMIT_FIELD;
+
```
Comment (ryantxu) [inline packages/grafana-data/src/transformations/transformers/limit.ts:27]: Totally untested, but maybe somehting like:
```suggestion
return data.map(frame => {
  if(frame.length > limitFieldMatch ) {
    return {
       ...frame,
       fields: frame.fields.map( f => {
           const vals = f.values.toArray();
           return {
              ...f,
              values: new ArrayVector(vals.slice(0,limitFieldMatch)),
           }
       },
       length: limitFieldMatch,
    }
  }
  return frame;
});
```

**Comment by josiahg** [comment]
@ryantxu who should I bring in to review this?

**Review by dprokop — APPROVED** [review]
@josiahg - works as expected and the code is good!

**Pull request merged**